packages/view/src/subjects/PrimitiveSubject.ts

```diff
@@ -27,10 +28,46 @@ export class PrimitiveSubject extends Subject<PrimitiveDef, MeshLike> {
 	protected attributes = new RefMapObserver<AccessorDef, BufferAttribute>('attributes', this._documentView);
 
 	constructor(documentView: DocumentViewSubjectAPI, def: PrimitiveDef) {
+		// Add Morph Targets
+		const geometry = new BufferGeometry();
+
+		// Convert geometry attributes
+		for (const accessor of def.listAttributes()) {
+			const array = accessor.getArray() as TypedArray;
+			geometry.setAttribute(accessor.getName(), new BufferAttribute(array, accessor.getElementSize()));
+		}
+
+		// Convert indices if available
+		if (def.getIndices()) {
+			const accessor = def.getIndices() as AccessorDef;
+			const array = accessor.getArray() as TypedArray;
+			geometry.setIndex(new BufferAttribute(array, 1));
+		}
+
+		// Convert morph targets
+		const targets = def.listTargets();
+		if (targets.length > 0) {
+			geometry.morphAttributes = {};
+
+			for (const [index, target] of targets.entries()) {
+				for (const semantic of target.listSemantics()) {
+					const attriName = semantic.toLowerCase();
+					if (!geometry.morphAttributes[attriName]) {
+						geometry.morphAttributes[attriName] = [];
+					}
+					const accessor = target.getAttribute(semantic) as AccessorDef;
+					const array = accessor.getArray() as TypedArray;
+					geometry.morphAttributes[attriName][index] = new BufferAttribute(array, accessor.getElementSize());
+				}
+			}
+
+			geometry.morphTargetsRelative = true;
+		}
```

Comment:
At some point, it will probably be necessary to create a `PrimitiveTargetSubject` class that handles listening for changes to each morph target definition, and emits events so that the BufferGeometry can be updated when things change.

It's OK with me if the morph targets are readonly (just one-time setup, no live updates)  for now though. Perhaps this code for adding targets to the geometry could just be moved into `PrimitiveSubject.createValue`, or a helper function, to keep the constructor concise?

Reply:
@donmccurdy For now, I've moved the code to `PrimitiveSubject.createValue`.